full_hashtable/hashtables.c

```diff
@@ -135,7 +135,27 @@ void hash_table_insert(HashTable *ht, char *key, char *value)
  */
 void hash_table_remove(HashTable *ht, char *key)
 {
+  unsigned int index = hash(key, ht->capacity);
+  if (ht->storage[index]->key != NULL && strcmp(ht->storage[index]->key, key) == 0)
+  {
+    if (index > 0)
+    {
+      if (ht->storage[index - 1] != NULL)
+      {
+        LinkedPair *previous_pair = ht->storage[index - 1];
```

Comment:
Creating a LinkedPair and holding it. This block of code seems a bit complicated to understand, consider adding a comment here